libethereum/Executive.cpp

```diff
@@ -515,6 +515,8 @@ void revertAccountChanges(State& _state, AccountRevertLog const& _changes)
 			_state.setNonce(_changes.address, 0);
 			_state.setCode(_changes.address, {});
 			_state.clearStorageChanges(_changes.address);
+			// Also untouch it because it classify for being cleared any more.
+			_state.untouch(_changes.address);
```

Comment:
I see.

Maybe we can here just move the `if (!_changes.existed) _state.untouch(_changes.address)` from lines 531-532 to outside of `if (_changes.isCreation)` (to line 534)

Then it's not repeated